pwiz_tools/Skyline/FileUI/PeptideSearch/ConverterSettingsControl.cs

```diff
@@ -0,0 +1,314 @@
+                        defaultDiaUmpireSettings[param.Key] = new AbstractDdaSearchEngine.Setting(param.Key, s);
+                        allDiaUmpireSettings[param.Key] = new AbstractDdaSearchEngine.Setting(param.Key, s);
+                        break;
+                    case int i32:
+                        defaultDiaUmpireSettings[param.Key] = new AbstractDdaSearchEngine.Setting(param.Key, i32);
+                        allDiaUmpireSettings[param.Key] = new AbstractDdaSearchEngine.Setting(param.Key, i32);
+                        break;
+                    case float r32:
+                        defaultDiaUmpireSettings[param.Key] = new AbstractDdaSearchEngine.Setting(param.Key, r32);
+                        allDiaUmpireSettings[param.Key] = new AbstractDdaSearchEngine.Setting(param.Key, r32);
+                        break;
+                    case double r64:
+                        defaultDiaUmpireSettings[param.Key] = new AbstractDdaSearchEngine.Setting(param.Key, r64);
+                        allDiaUmpireSettings[param.Key] = new AbstractDdaSearchEngine.Setting(param.Key, r64);
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
+                }
+            }
+
+            foreach (var param in _diaUmpireAdditionalSettings)
+                allDiaUmpireSettings[param.Key].Value = param.Value.Value;
+
+            // only non-default settings are kept in _diaUmpireAdditionalSettings
+            _diaUmpireAdditionalSettings = new Dictionary<string, AbstractDdaSearchEngine.Setting>();
+
+            Func<AbstractDdaSearchEngine.Setting, string> valueToString = setting =>
+            {
+                if (setting.Value is double)
+                    return Math.Round((double) setting.Value, 4).ToString(@"F");
+                return setting.Value.ToString();
+            };
+
+            Action<string, AbstractDdaSearchEngine.Setting> stringToValueIfNonDefault = (value, setting) =>
+            {
+                if (value.ToString() == valueToString(defaultDiaUmpireSettings[setting.Name]))
+                    return;
+                _diaUmpireAdditionalSettings[setting.Name] = setting;
+                setting.Value = value;
+            };
+
+            stringToValueIfNonDefault(EstimateBackground.ToString(), defaultDiaUmpireSettings[ESTIMATEBG]);
+
+            KeyValueGridDlg.Show(Resources.SearchSettingsControl_Additional_Settings,
+                allDiaUmpireSettings, valueToString, stringToValueIfNonDefault,
+                (value, setting) => setting.Validate(value));
+        }
+
+        private void UpdateSettingIfNonDefault<T>(IDictionary<string, AbstractDdaSearchEngine.Setting> settingStore,
+            string settingName, T newValue, T defaultValue) where T : IEquatable<T>
+        {
+            if (newValue.Equals(defaultValue))
+                return;
+
+            dynamic newValueDynamic = newValue;
```

Comment:
Doing constructor overload resolution at runtime with a generic parameter required this dynamic voodoo, which required adding the Microsoft.CSharp reference. Any problem with that @nickshulman ?